[DOCS-1639] Docs for CoreWeave BYOB on Multi-tenant Cloud (#1446)
wandb/docs: content/en/guides/hosting/data-security/secure-storage-connector.md

The following table shows the availability of BYOB at each scope for each W&B deployment type.

| W&B deployment type        | Instance level   | Team level | Additional information |

> |----------------------------|------------------|------------|------------------------|

Reviewer: Nit: The correct product name is "Amazon S3" not "AWS S3"

> | Dedicated Cloud            | &check;          | &check;    | Instance and team level BYOB are supported for CoreWeave AI Object Storage, AWS S3, GCP Storage, Microsoft Azure Blob Storage, and S3-compatible storage like [MinIO](https://github.com/minio/minio) hosted in your cloud or on-prem infrastructure. |

Author: ```suggestion
| Dedicated Cloud            | &check;          | &check;    | Instance and team level BYOB are supported for CoreWeave AI Object Storage, Amazon S3, GCP Storage, Microsoft Azure Blob Storage, and S3-compatible storage like [MinIO](https://github.com/minio/minio) hosted in your cloud or on-premises infrastructure. |
```

> | Multi-tenant Cloud         | Not Applicable   | &check;    | Team level BYOB is supported for CoreWeave AI Object Storage, AWS S3, and GCP Storage. W&B fully manages the default and only storage bucket for Microsoft Azure. |

Author: ```suggestion
| Multi-tenant Cloud         | Not Applicable   | &check;    | Team level BYOB is supported for CoreWeave AI Object Storage, Amazon S3, and GCP Storage. W&B fully manages the default and only storage bucket for Microsoft Azure. |
```

> | Self-Managed               | &check;          | &check;    | Instance and team level BYOB are supported for CoreWeave AI Object Storage, AWS S3, GCP Storage, Microsoft Azure Blob Storage, and S3-compatible storage like [MinIO](https://github.com/minio/minio) hosted in your cloud or infrastructure on your premises. |

Author: ```suggestion
| Self-Managed               | &check;          | &check;    | Instance and team level BYOB are supported for CoreWeave AI Object Storage, Amazon S3, GCP Storage, Microsoft Azure Blob Storage, and S3-compatible storage like [MinIO](https://github.com/minio/minio) hosted in your cloud or infrastructure on your premises. |
```

The following sections guide you through the process of setting up BYOB.

## Provision your bucket {#provision-your-bucket}

After [verifying availability]({{< relref "#availability-matrix" >}}), you are ready to provision your storage bucket, including its access policy and CORS. Select a tab to continue.

{{< tabpane text=true >}}
{{% tab header="CoreWeave" value="coreweave" %}}
<a id="coreweave-requirements"></a>**Requirements**:
- **Dedicated Cloud** or **Self-Hosted** v0.70.0 or newer, or **Multi-tenant Cloud**.
- A CoreWeave account with AI Object Storage enabled and with permission to create buckets, API access keys, and secret keys.
- Your W&B instance must be able to connect to CoreWeave network endpoints.

For details, see [Create a CoreWeave AI Object Storage bucket](https://docs.coreweave.com/docs/products/storage/object-storage/how-to/create-bucket) in the CoreWeave documentation.

1. **Multi-tenant Cloud**: Obtain your organization ID, which is required for your bucket policy.

>     1. Log in to [wandb.ai/](https://wandb.ai/).

Reviewer: (Thinking of recent changes) Did you want to change this to "Log in to the W&B App."?

>     1. Log in to [wandb.ai/](https://wandb.ai/).

Author: ```suggestion
    1. Log in to the [W&B App](https://wandb.ai/).
```

    1. In the left navigation, click **Create a new team**.

>     1. In the drawer that opens, copy the W&B organization ID, which is just above **Invite team members**.

Reviewer: ```suggestion
    1. In the drawer that opens, copy the W&B organization ID, which is located above **Invite team members**.
```

>     1. Leave this page open. You will use it to [configure W&B](#configure-wb).

Author: ```suggestion
    1. Leave this page open. You will use it to [configure W&B]({{< relref "#configure-byob" >}}).
```

    Replace `<cw-bucket>` with the CoreWeave bucket name and replace `<wb-org-id>` with the W&B organization ID you obtained in step 1.

    ```json
    {
      "Version": "2012-10-17",
      "Statement": [
      {
        "Sid": "AllowWandbUser",
        "Action": [
          "s3:GetObject*",
          "s3:GetEncryptionConfiguration",
          "s3:ListBucket",
          "s3:ListBucketMultipartUploads",
          "s3:ListBucketVersions",
          "s3:AbortMultipartUpload",
          "s3:DeleteObject",
          "s3:PutObject",
          "s3:GetBucketCORS",
          "s3:GetBucketLocation",
          "s3:GetBucketVersioning"
        ],
        "Effect": "Allow",
        "Resource": [
          "arn:aws:s3:::<cw-bucket>/*",
          "arn:aws:s3:::<cw-bucket>"
        ],
        "Principal": {

>           "CW": "arn:aws:iam::wandb:static/wandb-artifacts"

Reviewer 2: the gist i gave you was a bit old we changed it to wandb-integration
```suggestion
          "CW": "arn:aws:iam::wandb:static/wandb-integration"
```

        },
        "Condition": {
          "StringLike": {
            "wandb:OrgID": [
              "<wb-org-id>"
            ]
          }
        }
      },
      {

>         "Sid": "AllowUsersInOrg",

Reviewer 2: maybe add a comment here saying you don't need this unless you still want users in your org to be able to have access to this bucket.

Author: You can't comment in JSON, according to Google. I'll add the detail right after the policy. Thanks.

        "Action": "s3:*",
        "Effect": "Allow",
        "Resource": [
          "arn:aws:s3:::<cw-bucket>",
          "arn:aws:s3:::<cw-bucket>/*"
        ],
        "Principal": {
          "CW": "arn:aws:iam::b13ad0:*"
        }
      }]
    }
    ```

> 

Author: ```suggestion

The clause beginning with `"Sid": "AllowUsersInOrg"` grants users in your W&B organization direct access to the bucket. If you don't need this ability, you can omit the clause from your policy.
```

- If you plan to configure CoreWeave storage for the team, contact [support](mailto:support@wandb.com) to verify that your bucket is configured correctly in CoreWeave and to validate your team's configuration, since the storage details cannot be changed after the team is created.
{{% /alert %}}

Select your deployment type to continue.

{{< tabpane text=true >}}
{{% tab header="Dedicated Cloud / Self-Hosted" value="dedicated" %}}

> 1. If you’re connecting a W&B Team on **Dedicated Cloud** or **Self-Managed** to a cloud-native storage bucket in another cloud or to an S3-compatible storage bucket like MinIO, you **must** add the bucket path to the `GORILLA_SUPPORTED_FILE_STORES` environment variable and then restart W&B before following the rest of these steps to use the storage bucket for a team.

Reviewer: Suggestion : Add step 1 as a "Requirement"? Similar to the above.

Author: We chatted about this in Slack. This step is required for Self-Hosted or else you won't be able to do the rest of the steps. It's actually not possible on Dedicated Cloud, so I'll adjust it.
```suggestion
1. **Dedicated Cloud**: You **must** provide the bucket path to your account team so that they can add it to your instance's supported file stores before following the rest of these steps to use the storage bucket for a team.
1. **Self-Managed**: You **must** add the bucket path to your the `GORILLA_SUPPORTED_FILE_STORES` environment variable and then restart W&B before following the rest of these steps to use the storage bucket for a team.
```

- If you plan to configure CoreWeave storage for the team, contact [support](mailto:support@wandb.com) to verify that your bucket is configured correctly in CoreWeave and to validate your team's configuration, since the storage details cannot be changed after the team is created.
{{% /alert %}}

Select your deployment type to continue.

{{< tabpane text=true >}}
{{< tab header="Dedicated Cloud / Self-Hosted" value="dedicated" >}}

> 1. If you’re connecting to a cloud-native storage bucket in another cloud or to an S3-compatible storage bucket like MinIO for team-level BYOB in your Dedicated cloud or Self-managed instance, you **must** add the bucket path to the `GORILLA_SUPPORTED_FILE_STORES` environment variable and then restart W&B, before following the rest of these steps to use the storage bucket for a team.

Reviewer 2: doesn't seem like this is properly formatted in the preview?
<img width="928" height="727" alt="Screenshot 2025-07-10 at 17 03 30" src="https://github.com/user-attachments/assets/229b705a-1681-4405-bff0-578a24cd522d" />

1. Click **Bucket location**.
1. To use an existing bucket, select it from the list. To add a new bucket, click **Add bucket** at the bottom, then provide the bucket's details.

    Click **Cloud provider** and select **CoreWeave**, **AWS**, **GCP**, or **Azure**.
    

>     On **Self-Managed**, if the cloud provider is not listed, ensure that you followed step 1 to add the bucket path to the `GORILLA_SUPPORTED_FILE_STORES` environment variable. If no buckets from that provider are included in the environment variable, that provider is not listed.

Author: ```suggestion
    If the cloud provider is not listed, ensure that you have followed step 1 to add the bucket path to the supported file stores for your instance. If the storage provider is still not listed, [contact support](mailto:support@wandb.ai) for assistance.
```

    Click **Cloud provider** and select **CoreWeave**, **AWS**, **GCP**, or **Azure**. CoreWeave is not yet available for teams on Multi-tenant Cloud. If the cloud provider is not listed, ensure that you followed step 1 to add the bucket path to the `GORILLA_SUPPORTED_FILE_STORES` environment variable. If no buckets from that provider are include in the environment variable, that provider is not listed.
    -  Specify the bucket.
    
        - For **CoreWeave**, provide only the bucket name.
        - For Amazon S3, GCP, or S3-compatible storage, provide the full bucket path you [determined earlier](#determine-the-storage-address).
        - For Azure on W&B Dedicated or Self-Managed, set **Account name** to the Azure account and **Container name** to the Azure blob storage container.
    - Optionally:
      - If applicable, set **Path** to the bucket sub-path.
      - **AWS**: Set **KMS key ARN** to the ARN of your KMS encryption key.
      -  **Azure**: If applicable, specify values for **Tenant ID** and **Managed Identity Client ID**.

>      - Invite members to the team. In **Invite team members**, specify a comma-separated list of email addresses. Otherwise, you can invite members to the team after it is created.

Reviewer 2: want to check if this is a mistake as for multi-tenant cloud there is no steps involving `GORILLA_SUPPORTED_FILE_STORES`?

Author: It's a mistake, I will fix it shortly. Thanks.

Author: Fixed, I believe.
